### ISSUE TITLE
showIncludes outputs to stdout

### ISSUE THREAD

**Issue by orgads** [MicrosoftDocs/cpp-docs#3785]
The doc explicitly states:

> The `/showIncludes` option emits to `stderr`, not `stdout`.

But that's wrong.

test.cpp:
```
#include <stdio.h>
int main() {}
```

Running cl, redirecting `stdout` to NUL:
```
> cl /showIncludes test.cpp > NUL
Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30139 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.
```

Without redirect:
```
> cl /showIncludes test.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30139 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

test.cpp
Note: including file: C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\stdio.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt.h
Note: including file:   C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\vcruntime.h
Note: including file:    C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\shared\sal.h
Note: including file:     C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\concurrencysal.h
Note: including file:    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\vadefs.h
Note: including file:  C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_wstdio.h
Note: including file:   C:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt\corecrt_stdio_config.h
Microsoft (R) Incremental Linker Version 14.29.30139.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:test.exe
test.obj
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3940639d-19f6-f22a-7a71-cf33016db99a
* Version Independent ID: 66515261-fe4e-1f96-a54c-f1b96f61bc77
* Content: [/showIncludes (List include files)](https://docs.microsoft.com/en-us/cpp/build/reference/showincludes-list-include-files?view=msvc-170)
* Content Source: [docs/build/reference/showincludes-list-include-files.md](https://github.com/Microsoft/cpp-docs/blob/main/docs/build/reference/showincludes-list-include-files.md)
* Product: **visual-cpp**
* Technology: **cpp-tools**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by corob-msft** [comment]
@orgads 
It's hard to argue with the evidence. I went looking for the preprocessor code that prints the include files, and it does in fact appear to send the output to `stderr` as claimed - but there's more to the story. It calls an output function that performs some checks to see whether the output is going to the console, to a file, or to the IDE. Since the invoked function sends the output to `stderr`, that counts as the console - and output is printed using `_cputws`. So, it doesn't get sent to `stderr` after all. Given the evidence, it looks like once upon a time this output really did go to `stderr` as described, but sometime in the misty past, it got intercepted, and no longer does. Only the copyright banner goes to `stderr`.
I've updated the article accordingly, and it should be visible tomorrow.
